fem/pfespace.cpp

```diff
@@ -3147,9 +3199,9 @@ static void SetSubVector(const int N,
                          const Array<int> &indices,
                          const Vector &in, Vector &out)
 {
-   auto y = out.Write();
-   const auto x = in.Read();
-   const auto I = indices.Read();
+   auto y = out.HostWrite();
+   const auto x = in.HostRead();
+   const auto I = indices.HostRead();
    MFEM_FORALL(i, N, y[I[i]] = x[i];);
 }
```

Comment:
I am confused by this too -- do we want the above assignment on the host or on the device? In the former case, `MFEM_FORALL` may look confusing, in the latter case the `HostRead` is confusing...